src/main/java/io/zksync/signer/DefaultEthSigner.java

```diff
@@ -80,8 +80,23 @@ public static DefaultEthSigner fromRawPrivateKey(Web3j web3j, String rawPrivateK
         return new DefaultEthSigner(new RawTransactionManager(web3j, credentials), credentials);
     }
 
+    public static DefaultEthSigner fromMnemonicEIP1271(Web3j web3j, String mnemonic, String contractAddress) {
+        Credentials credentials = generateCredentialsFromMnemonic(mnemonic, 0);
+        return new DefaultEthSigner(new RawTransactionManager(web3j, credentials), credentials, contractAddress);
+    }
+
+    public static DefaultEthSigner fromMnemonicEIP1271(Web3j web3j, String mnemonic, int accountIndex, String contractAddress) {
+        Credentials credentials = generateCredentialsFromMnemonic(mnemonic, accountIndex);
+        return new DefaultEthSigner(new RawTransactionManager(web3j, credentials), credentials, contractAddress);
+    }
+
+    public static DefaultEthSigner fromRawPrivateKeyEIP1271(Web3j web3j, String rawPrivateKey, String contractAddress) {
```

Comment:
How EIP1271 can be from raw private key? These arent usually created from mnemonics either

Reply:
Basically [EIP1271](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1271.md) describes only interface of smart contract (called ERC1271). Any smart-contract that implements ERC1271 can verify signatures in any required way. In this case library creates Signer that will make signatures using `rawPrivateKey` and verify signed data with ERC1271 compliant contract deployed at `contractAddress`